sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/FromQuery.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Azure.DigitalTwins.Core.Queries.QueryBuilders;
+
+namespace Azure.DigitalTwins.Core.QueryBuilder
+{
+    /// <summary>
+    /// Query object that already contains a select clause.
+    /// </summary>
+    public class FromQuery : QueryBase
+    {
+        private WhereQuery _innerQuery;
+        private AdtQueryBuilder _parent;
+        private IList<FromClause> _clauses;
+
+        internal FromQuery(AdtQueryBuilder parent, WhereQuery selectPart)
+        {
+            _parent = parent;
+
+            // TODO -- change to just a singular from clause
```

Comment:
you can make this happen, you probably don't need a list of clauses at this point.